src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -2677,14 +2677,17 @@ MethodTableBuilder::EnumerateClassMethods()
     if ((DWORD)MAX_SLOT_INDEX <= cMethAndGaps)
         BuildMethodTableThrowException(IDS_CLASSLOAD_TOO_MANY_METHODS);
 
-    bmtMethod->m_cMaxDeclaredMethods = (SLOT_INDEX)cMethAndGaps;
-
-    if (g_pConfig->RuntimeAsync())
+    // In a worst case the number of declared methods can double
+    // as each async method may have two variants.
+    // The method count is typically a modest number though.
+    // We will reserve twice the size for the builder, up to the max, just in case.
+    DWORD cMethUpperBound = cMethAndGaps * 2;
+    if (cMethUpperBound >= (DWORD)MAX_SLOT_INDEX)
```

Comment:
```suggestion
    if ((DWORD)MAX_SLOT_INDEX <= cMethUpperBound)
```
So that it is same style as ` if ((DWORD)MAX_SLOT_INDEX <= cMethAndGaps)`  a few lines above